pkg/utils/kubernetes/client/client.go

```diff
@@ -239,17 +242,17 @@ func DefaultCleaner() Cleaner {
 }
 
 // Clean deletes and optionally finalizes resources that expired their termination date.
-func (o *cleaner) Clean(ctx context.Context, c client.Client, obj runtime.Object, opts ...CleanOptionFunc) error {
+func (o *cleaner) Clean(ctx context.Context, k8sInterface kubernetes.Interface, obj runtime.Object, opts ...CleanOptionFunc) error {
```

Comment:
Please don't change the interface - rather either
* create a new implementation of the `Cleaner` interface that supports cleaning a namespace
-> Here, you could re-use the existing `cleaner`. Once you have this, it's easy to re-use the existing `CleanOps`.
* write a single function that handles finalizing namespaces and have this completely separate from the generic `Cleaner`

Reply:
Introduced `Finalizer` interface. Interface changes are now reverted.